build_image.sh

```diff
@@ -200,6 +200,11 @@ elif [ "$IMAGE_TYPE" = "aboot" ]; then
     generate_device_list ".platforms_asic"
     zip -g $OUTPUT_ABOOT_IMAGE .platforms_asic
 
+    echo "sonic_fips=0" > kernel-cmdline
+    [ "$ENABLE_FIPS" == "y" ] && echo "sonic_fips=1" > kernel-cmdline
+    zip -g $OUTPUT_ABOOT_IMAGE kernel-cmdline
+    rm kernel-cmdline
```

Comment:
Your solution using `kernel-cmdline` will work.
One note, even though the kernel handles multiple definitions of the same parameter properly, it's probably not worth crowding the cmdline when enabling fips (will append `sonic_fips=0 sonic_fips=1` which can be confusing)

If the plan is to configure FIPS only at build time I would suggest making a modification to `boot0.j2` and add a new variable `enable_fips` set via Jinja templating to `true` or `false`.
Then adding some logic in the `boot0` code to call `cmdline_add` in  the `write_image_specific_config` function
https://github.com/Azure/sonic-buildimage/blob/f6927606b3720d4f526b9e734b4431f012d21ee3/files/Aboot/boot0.j2#L617
This would prevent anyone from disabling `sonic_fips` by changing the cmdline in the context of secureboot.
However if the plan is to be able to disable fips at runtime or for a next reboot, your solution works better.

Reply:
@Staphylo , it makes the user have chance to enable or disable it in the next reboot. ENABLE_FIPS=1 will enable the fips by default, the default value is "n", not enabled. If want to enable it, we need to set the sonic_fips=1 in kernel-cmdline after installed or upgraded.